### PR TITLE
SOR Configuration changes to facilitate data driven UI

### PR DESCRIPTION
**Pull request by Bhavesh-Patel**

### CODE REVIEW

**Comment by Bhavesh-Patel** [comment]
[Implement data driven UI for triage data steps (Backend)](https://trello.com/c/JDzuL9BD/232-implement-data-driven-ui-for-triage-data-steps-backend)

File: HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Linq;
```
Comment (agnieszkazbikowska-madetech) [inline HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs:2]: It seems that `using System.Linq;` is an unused import.

File: HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                result.Add(new SorConfiguration { Display = "display", Value = earlyExitValue });
+            }
+
+            return result;
+        }
+
+        public static TheoryData<SorConfiguration> ValidSorConfigurationsWithEitherSorCodeOrOptions() =>
+            new()
+            {
+                new SorConfiguration { Display = "display", Value = "value", SorCode = "sor code" },
```
Comment (agnieszkazbikowska-madetech) [inline HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs:37]: I'm wondering if we could use a variable for the repeated values here & then use that variable?

File: HousingRepairsOnlineApi.Tests/HelpersTests/ServiceCollectionExtensionsTests.cs [hunk 3]
```diff
@@ -62,35 +78,267 @@ public void GivenValidSorConfigPathArgument_WhenAddingSoREngineToServices_ThenSo
...
+                    }]
+                }]"
+                },
             };
-            yield return new object[] { new JsonReaderException(), @"
+        }
+
+        public static TheoryData<IEnumerable<SorConfiguration>, IEnumerable<RepairTriageOption>> ValidSorConfigurationForCreatingJourneyRepairTriageOptions()
+        {
+            var value = "value1";
```
Comment (agnieszkazbikowska-madetech) [inline HousingRepairsOnlineApi.Tests/HelpersTests/ServiceCollectionExtensionsTests.cs:176]: I've got prompts here from my code editor to change these into `constant`s.

Comment (Bhavesh-Patel) [reply HousingRepairsOnlineApi.Tests/HelpersTests/ServiceCollectionExtensionsTests.cs]: Extracted to field constants

File: HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs (hunk 2 above)
Comment (Bhavesh-Patel) [reply HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs]: Extracted to field constants

File: HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs (hunk 1 above)
Comment (Bhavesh-Patel) [reply HousingRepairsOnlineApi.Tests/Domain/SorConfigurationExtensionsTests.cs]: Removed

**Review by agnieszkazbikowska-madetech — APPROVED** [review]

**Pull request merged**